src/monattack.cpp

```diff
@@ -4860,3 +4860,96 @@ bool mattack::dodge_check(monster *z, Creature *target){
     }
     return false;
 }
+
+void mattack::ink_jet(monster *z, int index) {
+
+	if (!z->can_act()) {
+		return;
+	}
+
+	int jet_range =  5;
+
+	// TODO: make the creature utilise this ability then escaping the threat
+	Creature *target = z->attack_target();
+
+	if (target == nullptr || rl_dist(z->pos(), target->pos()) > jet_range || !z->sees(*target)) {
+		return;
+	}
+
+	std::vector<tripoint> line = g->m.find_clear_path(z->pos(), target->pos());
+
+	z->reset_special(index);
+
+	z->moves -= 50;
+
+	// Notify the player, if he sees the use of this ability
+	bool u_see = g->u.sees(*z);
+	if (u_see) {
+		if (target->is_player()) {
+			add_msg(m_bad, _("The %s squirts a jet of black ink at you!"), z->name().c_str());
+		}
+		else if (target->is_npc()) {
+
+			if (target->attitude_to(g->u) == Creature::A_FRIENDLY) {
+				add_msg(m_bad, _("The %s squirts a jet of black ink at <npcname>!"),
+					z->name().c_str(), target->disp_name().c_str());
+			}
+			else {
+				add_msg(m_warning, _("The %s squirts a jet of black ink at <npcname>!"),
+					z->name().c_str(), target->disp_name().c_str());
+			}
+		}
+		else if (target->attitude_to(g->u) == Creature::A_FRIENDLY) {
+			add_msg(m_bad, _("The %1$s squirts a jet of black ink at %2$s!"),
+				z->name().c_str(), target->disp_name().c_str());
+		}
+		else {
+			add_msg(m_warning, _("The %1$s squirts a jet of black ink at %2$s!"),
+				z->name().c_str(), target->disp_name().c_str());
+		}
+	}
```

Comment:
Too many cases here just to determine the color of the message.
You could achieve it all with:
```
if( u_see || target->is_player() ) {
    const auto msg_type = (target->attitude_to(g->u) == Creature::A_FRIENDLY ? m_bad : m_warning);
    target->add_msg_player_or_npc( msg_type, _("The %s squirts a jet of black ink at you!"), _("The %s squirts a jet of black ink at <npcname>!") );
}
```

This should handle all of the cases - monsters, NPCs and the player, both friendly and unfriendly.